Sometimes crashes with "android.app.ForegroundServiceStartNotAllowedException: Time limit already exhausted for foreground service type unknown"

**Platform**
- [ ] iOS (version: )
- [*] Android (version: GrafeneOS 16)

**SyncUp version**
1.1.14

**Describe the bug**
Sometimes android shows message like "App keeps crashing".
If runned from launcher screen seems to work fine.

**Steps to reproduce**
Do not know how to reproduce

**Logs**


```
type: crash
osVersion: google/caiman/caiman:16/BP4A.260205.002/2026060101:user/release-keys
package: com.siddarthkay.syncup:29664507, targetSdk 36
process: com.siddarthkay.syncup
processUptime: 537 + 566 ms
installer: org.fdroid.fdroid

java.lang.RuntimeException: Unable to create service com.siddarthkay.syncup.SyncthingService
        at android.app.ActivityThread.handleCreateService(ActivityThread.java:5593)
        at android.app.ActivityThread.-$$Nest$mhandleCreateService(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2745)
        at android.os.Handler.dispatchMessage(Handler.java:132)
        at android.os.Looper.dispatchMessage(Looper.java:333)
        at android.os.Looper.loopOnce(Looper.java:263)
        at android.os.Looper.loop(Looper.java:367)
        at android.app.ActivityThread.main(ActivityThread.java:9332)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:566)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:837)
Caused by: android.app.ForegroundServiceStartNotAllowedException: Time limit already exhausted for foreground service type unknown
        at android.app.ForegroundServiceStartNotAllowedException$1.createFromParcel(ForegroundServiceStartNotAllowedException.java:54)
        at android.app.ForegroundServiceStartNotAllowedException$1.createFromParcel(ForegroundServiceStartNotAllowedException.java:50)
        at android.os.Parcel.readParcelableInternal(Parcel.java:5282)
        at android.os.Parcel.readParcelable(Parcel.java:5263)
        at android.os.Parcel.createExceptionOrNull(Parcel.java:3377)
        at android.os.Parcel.createException(Parcel.java:3366)
        at android.os.Parcel.readException(Parcel.java:3342)
        at android.os.Parcel.readException(Parcel.java:3284)
        at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:7512)
        at android.app.Service.startForeground(Service.java:878)
        at com.siddarthkay.syncup.SyncthingService.promoteToForeground(SyncthingService.kt:364)
        at com.siddarthkay.syncup.SyncthingService.onCreate(SyncthingService.kt:118)
        at android.app.ActivityThread.handleCreateService(ActivityThread.java:5582)
        ... 10 more
```

**Screenshots**
None

--- a/mobile-app/android/app/src/main/AndroidManifest.xml
+++ b/mobile-app/android/app/src/main/AndroidManifest.xml
@@ -6,7 +6,6 @@
   <uses-permission android:name="android.permission.CHANGE_WIFI_MULTICAST_STATE"/>
   <uses-permission android:name="android.permission.WAKE_LOCK"/>
   <uses-permission android:name="android.permission.FOREGROUND_SERVICE"/>
-  <uses-permission android:name="android.permission.FOREGROUND_SERVICE_DATA_SYNC"/>
   <uses-permission android:name="android.permission.FOREGROUND_SERVICE_SPECIAL_USE"/>
   <uses-permission android:name="android.permission.POST_NOTIFICATIONS"/>
   <uses-permission android:name="android.permission.READ_MEDIA_IMAGES"/>
@@ -60,7 +59,7 @@
         <data android:mimeType="*/*"/>
       </intent-filter>
     </activity>
-    <service android:name=".SyncthingService" android:foregroundServiceType="dataSync|specialUse" android:exported="false">
+    <service android:name=".SyncthingService" android:foregroundServiceType="specialUse" android:exported="false">
       <property android:name="android.app.PROPERTY_SPECIAL_USE_FOREGROUND_SERVICE_SUBTYPE" android:value="continuous_file_sync_daemon"/>
     </service>
     <receiver android:name=".AppConfigReceiver" android:exported="true">

--- a/mobile-app/android/app/src/main/java/com/siddarthkay/syncup/SyncthingService.kt
+++ b/mobile-app/android/app/src/main/java/com/siddarthkay/syncup/SyncthingService.kt
@@ -115,7 +115,11 @@ class SyncthingService : Service() {
         super.onCreate()
         Log.i(TAG, "onCreate")
         createNotificationChannel()
-        promoteToForeground()
+        if (!promoteToForeground()) {
+            Log.w(TAG, "promoteToForeground failed; stopping service")
+            stopSelf()
+            return
+        }
         acquireMulticastLock()
         startDaemon()
         registerNetworkCallback()
@@ -352,23 +356,23 @@ class SyncthingService : Service() {
 
     private fun buildNotification(): Notification = buildNotificationStandalone(this)
 
-    private fun promoteToForeground() {
+    // Returns true once the service is running in the foreground.
+    private fun promoteToForeground(): Boolean {
         val notification = buildNotification()
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
-            // dataSync alone is rejected from BOOT_COMPLETED on Android 14+
-            // (per-type allowlist for background broadcast contexts). OR'ing
-            // specialUse satisfies the boot path while preserving dataSync's
-            // semantics for foreground / share / activity launches. Manifest
-            // declares both types and the required PROPERTY_SPECIAL_USE_*
-            // subtype.
-            startForeground(
-                NOTIFICATION_ID,
-                notification,
-                ServiceInfo.FOREGROUND_SERVICE_TYPE_DATA_SYNC or
+        return try {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+                startForeground(
+                    NOTIFICATION_ID,
+                    notification,
                     ServiceInfo.FOREGROUND_SERVICE_TYPE_SPECIAL_USE,
-            )
-        } else {
-            startForeground(NOTIFICATION_ID, notification)
+                )
+            } else {
+                startForeground(NOTIFICATION_ID, notification)
+            }
+            true
+        } catch (e: Exception) {
+            Log.e(TAG, "startForeground failed", e)
+            false
         }
     }
 }

